src/log.ts

```diff
@@ -0,0 +1,96 @@
+import { default as request } from 'request';
+import { Writable } from 'stream';
+
+import { KubeConfig } from './config';
+
+export interface LogOptions {
```

Comment:
This is taken over from #214 as it seems as a nice way to avoid overly-parameterized methods